tests/test_models/test_sarimax_model.py

```diff
@@ -229,10 +231,55 @@ def test_components_sum_up_to_target(
     model.fit(train, regressors)
 
     components_method = getattr(model, components_method_name)
+    predict_method = getattr(model, predict_method_name)
 
     pred_df = train if in_sample else test
 
-    pred = model.predict(pred_df, prediction_interval=False, quantiles=[])
+    pred = predict_method(df=pred_df, prediction_interval=False, quantiles=[])
     components = components_method(df=pred_df)
 
     np.testing.assert_allclose(np.sum(components.values, axis=1), np.squeeze(pred))
+
+
+def test_predict_components_of_subset_sum_up_to_target(dfs_w_exog):
+    train, _ = dfs_w_exog
+
+    model = _SARIMAXAdapter()
+    model.fit(train, ["f1", "f2"])
+
+    pred_df = train.iloc[5:-5]
+
+    components = model.predict_components(df=pred_df)
+    pred = model.predict(df=pred_df, prediction_interval=False, quantiles=[])
+
+    np.testing.assert_allclose(np.sum(components.values, axis=1), np.squeeze(pred))
+
+
+def test_forecast_components_of_subset_error(dfs_w_exog):
```

Comment:
Shouldn't it work in this case?  I mean if we have long enough regressors?

Reply:
In the case of SARIMAX forecast decomposition, we need to estimate full horizon. Here we cut one timestamp from the beginning and providing 3 points, while 4 expected.

Reply:
Does it works properly without regressors? If so add test for it

Reply:
Actually without regressors we don't have any decomposition as we will return only SARIMA component.